test/vtcomposite-localize-class.test.js

```diff
@@ -5,259 +5,825 @@ const { vtinfo } = require('./test-utils.js');
+        name: 'admin',
+        features: [
+          {
+            id: 10,
+            tags: [
+              0, 0, // _mbx_worldview
+              1, 1  // _mbx_class
+            ],
+            type: 1, // point
+            geometry: [9, 55, 38]
+          }
+        ],
+        keys: ['_mbx_worldview', '_mbx_class'],
+        values: [
+          { string_value: 'US' },
+          { string_value: 'affogato' },
+        ],
+        extent: 4096
+      }
+    ]
+  }).buffer;
+
+  const params = {
+    buffer: feature,
+    worldviews: ['JP']
+  };
+
+  localize(params, (err, vtBuffer) => {
+    assert.ifError(err);
+    const tile = vtinfo(vtBuffer);
+    assert.deepEqual(tile.layers, {}, 'has no feature');
+    assert.end();
+  });
+});
+
+test('[localize class] requesting localized worldview; feature with incompatible worldview key in "all" worldview', (assert) => {
+  const feature = mvtFixtures.create({
+    layers: [
+      {
+        version: 2,
+        name: 'admin',
+        features: [
+          {
+            id: 10,
+            tags: [
+              0, 0, // _mbx_worldview
+              1, 1, // _mbx_class
+              2, 0, // worldview
+              3, 2  // class
+            ],
+            type: 1, // point
+            geometry: [9, 55, 38]
+          }
+        ],
+        keys: ['_mbx_worldview', '_mbx_class', 'worldview', 'class'],
+        values: [
+          { string_value: 'all' },
+          { string_value: 'affogato' },
+          { string_value: 'fancy_affogato' }
+        ],
+        extent: 4096
+      }
+    ]
+  }).buffer;
+
+  const params = {
+    buffer: feature,
+    worldviews: ['US']
+  };
+
+  localize(params, (err, vtBuffer) => {
+    assert.ifError(err);
+    const tile = vtinfo(vtBuffer);
+    assert.ok('admin' in tile.layers, 'has admin layer');
+    assert.equal(tile.layers.admin.length, 1, 'has one feature');
+    assert.deepEqual(tile.layers.admin.feature(0).properties, { worldview: 'all', class: 'affogato' }, 'expected properties');
+    assert.end();
+  });
+});
+
+test('[localize class] requesting localized worldview; feature with incompatible worldview key in the requested worldview', (assert) => {
+  const feature = mvtFixtures.create({
+    layers: [
+      {
+        version: 2,
+        name: 'admin',
+        features: [
+          {
+            id: 10,
+            tags: [
+              0, 0, // worldview
+              1, 1  // class
+            ],
+            type: 1, // point
+            geometry: [9, 55, 38]
+          }
+        ],
+        keys: ['worldview', 'class'],
+        values: [
+          { string_value: 'US' },
+          { string_value: 'fancy_affogato' }
+        ],
+        extent: 4096
+      }
+    ]
+  }).buffer;
+
+  const params = {
+    buffer: feature,
+    worldviews: ['US']
+  };
+
+  localize(params, (err, vtBuffer) => {
+    assert.ifError(err);
+    const tile = vtinfo(vtBuffer);
+    assert.deepEqual(tile.layers, {}, 'has no feature');
```

Comment:
just to confirm, we're dropping this feature because we anticipate another feature with `_mbx_worldview` to exist with the value assigned as `US` ?

Reply:
Yes.